### PR TITLE
Switch to API v4

### PR DESCRIPTION
**Pull request by coder-hugo**

### CODE REVIEW

File: src/main/action/index.ts [hunk 1]
```diff
@@ -1,34 +1,55 @@
...
-    }
-  })();
+  const branch = await new Retryable<BranchOverview>(
+    async (retry, lastRetry) =>
+      await client.branchApi
+        .getBranches('me', configuration.projectId, { name: configuration.branchName })
+        .then(({ data }) => {
+          if (data.total === 0 || data.values[0].deploymentCount === 0) {
+            if (lastRetry) {
+              return data.values[0];
```
Comment (MM53) [inline src/main/action/index.ts:18]: This will fail if values is empty. Use a default object instead?

File: src/main/action/index.ts [hunk 2]
```diff
@@ -1,34 +1,55 @@
...
     warning('The deployment is disabled for this branch');
-    return { 'deployment-enabled': false };
+    return {
+      'deployment-enabled': false,
+    };
   }
 
-  const branch = await deployNowApi.getBranch(configuration.projectId, branchOverview.id);
+  if (branch.deploymentCount > 1) {
+    throw new Error("This action doesn't support multi deployments");
```
Comment (MM53) [inline src/main/action/index.ts:36]: Add link how to upgrade?

File: src/test/index.spec.ts [hunk 3]
```diff
@@ -2,132 +2,156 @@ import { expect } from 'chai';
...
+    sinon.replace(AxiosIonosSpaceBranchApiClient.prototype, 'getBranches', branchesStub);
 
-    const projectInfo = await retrieveProjectInfo({ apiKey: '', branchName: 'staging', projectId, serviceHost: '' });
-    const expected = {
-      'deployment-enabled': false,
-    };
+    try {
+      await retrieveProjectInfo({ apiKey: '', branchName: 'staging', projectId, serviceHost: '' });
+    } catch (e) {
+      expect(e).to.be.instanceof(Error);
```
Comment (MM53) [inline src/test/index.spec.ts:84]: Use `expect.to.throw` https://www.chaijs.com/api/bdd/#method_throw

File: src/main/action/index.ts (hunk 2 above)
Comment (coder-hugo) [reply src/main/action/index.ts:36]: If somebody sees this error he has done something strange as it's not possible to create an additional deployment for a project if its workflow isn't migrated.

**Review by MM53 — APPROVED** [review]

**Pull request merged**